zk/txpool/pool_xlayer.go

```diff
@@ -45,19 +53,33 @@ func (p *TxPool) checkWhiteAddr(addr common.Address) bool {
 	return false
 }
 
-func (p *TxPool) isFreeClaimAddr(senderID uint64) bool {
+func (p *TxPool) SetGpCacheForXLayer(gpCache GPCache) {
+	p.gpCache = gpCache
+}
+
+func (p *TxPool) checkFreeGasExAddress(senderID uint64) bool {
```

Comment:
Keep your function naming consistent
```suggestion
func (p *TxPool) checkFreeGasExAddr(senderID uint64) bool {
```